Source/WebCore/platform/graphics/cairo/PathCairo.cpp

```diff
@@ -328,148 +205,168 @@ void Path::addArcTo(const FloatPoint& p1, const FloatPoint& p2, float radius)
     if (orth_p1p0.y() < 0.f)
         sa = 2 * piDouble - sa;
 
-    // anticlockwise logic
-    bool anticlockwise = false;
+    // clockwise logic
+    auto direction = RotationDirection::Clockwise;
 
     float factor_p1p2 = tangent / p1p2_length;
     FloatPoint t_p1p2((p1.x() + factor_p1p2 * p1p2.x()), (p1.y() + factor_p1p2 * p1p2.y()));
-    FloatPoint orth_p1p2((t_p1p2.x() - p.x()),(t_p1p2.y() - p.y()));
+    FloatPoint orth_p1p2((t_p1p2.x() - p.x()), (t_p1p2.y() - p.y()));
     float orth_p1p2_length = sqrtf(orth_p1p2.x() * orth_p1p2.x() + orth_p1p2.y() * orth_p1p2.y());
     float ea = acos(orth_p1p2.x() / orth_p1p2_length);
     if (orth_p1p2.y() < 0)
         ea = 2 * piDouble - ea;
     if ((sa > ea) && ((sa - ea) < piDouble))
-        anticlockwise = true;
+        direction = RotationDirection::Counterclockwise;
     if ((sa < ea) && ((ea - sa) > piDouble))
-        anticlockwise = true;
+        direction = RotationDirection::Counterclockwise;
+
+    cairo_line_to(platformPath(), t_p1p0.x(), t_p1p0.y());
 
-    cairo_line_to(m_path.get(), t_p1p0.x(), t_p1p0.y());
+    addArc(p, radius, sa, ea, direction);
+
+    m_elementsStream = nullptr;
```

Comment:
Is this an existing issue, where including an arc in a path will cause the existing `m_elements` to be wrong since it's not including the arc? (And this is an improvement, but still not exactly correct, since we fall back to having no element stream at all in this case?)

Also, do we need to assign null here? `addArc` itself does that.